data/World/Overworld.json

```diff
@@ -1492,114 +1492,351 @@
         }
     },
     {
-        "region_name": "DMC Upper Nearby",
+        "region_name": "DMC Near Death Mountain Summit",
         "scene": "Death Mountain Crater",
         "hint": "Death Mountain Crater",
+        "events": {
+            "Smash Upper DMC Boulder": "can_use(Megaton_Hammer)"
+        },
         "exits": {
-            "DMC Upper Local": "can_use(Goron_Tunic)",
+            # The last couple lines of some exits from this region are based on going via the warp pad area.
             "Death Mountain Summit": "True",
-            "DMC Upper Grotto": "here(can_blast_or_smash)"
+            "GC Darunias Chamber": "
+                is_adult and (logic_fewer_tunic_requirements or Goron_Tunic) and 
+                (Hover_Boots or (Megaton_Hammer and 
+                (logic_crater_upper_to_lower or 'Smash Boulder Near Ladder')) or 
+                (Goron_Tunic and Longshot and (can_live_dmg(1) or 
```

Comment:
My concern around the full entrance->entrance logic is the maintainability; if we change how some traversal is done, or add a new trick, or whatever, we have to check every single region/entrance combo to see what needs to be updated. We'd avoided that previously by having the weird local/nearby stuff, but the repetition here makes me think we can define some LogicHelpers.json shortcuts, like something for
`(Goron_Tunic and Longshot and (can_live_dmg(1) or (Fairy and not entrance_shuffle) or can_use(Nayrus_Love)))`. That would be great because the name would be descriptive of what movement it refers to... e.g. `dmc_from_summit_via_warp_pad`, and then entrance rule reads like "have tunic or ftr, and then go via the ladder area or via the warp pad".

Reply:
There is just such a trick waiting in the wings. You can go upper to lower crater with no additional items with some weird roll jump. I'm not sure how much having logic helpers would help me implement this b/c the executing the trick takes some time, so it's gonna affect whether the various routes need tunic.